hcl2template/testdata/variables/variables_in_locals.pkr.hcl

```diff
@@ -0,0 +1,19 @@
+variable "name_prefix" {
+  default = "foo"
+}
+
+locals {
+  default_name_prefix = "${var.project_name}-web"
+  name_prefix         = "${var.name_prefix != "" ? var.name_prefix : local.default_name_prefix}"  // works
+  foo  = "${local.default_name_prefix}"
+  bar  = local.default_name_prefix
+}
+
+locals {
+  for_var  = "${local.default_name_prefix}"
+  bar_var  = local.default_name_prefix
+}
+
+variable "project_name" {
+  default = "test"
+}
```

Comment:
Nice test file ! 

```suggestion
}

```

Reply:
Do you think it would be a good idea to create a folder with a complicated test case ?

Something like:
```hcl
#b.pkr.hcl
locals {
  for_var  = "${local.name_prefix}"
  bar_var  = [
    local.for_var,
    local.foo,
    var.name_prefix,
  ]
}

#y.pkr.hcl
locals {
  name_prefix = "${var.name_prefi}"
  foo         = "${local.name_prefix}"
  bar         = local.name_prefix
}

#y.pkr.hcl
variable "name_prefix" {
  default = "foo"
}
```